File: Stack/Python/code.py
class Stack:
     def __init__(self):
         self.items = []

     def isEmpty(self):
         return self.items == []

     def push(self, item):
         self.items.append(item)

     def pop(self):
         return self.items.pop()

     def peek(self):
         return self.items[len(self.items)-1]

     def size(self):
         return len(self.items)
s=Stack()

print(s.isEmpty())
s.push(4)
s.push(5)
print(s.peek())
s.push(6)
s.push(10)
s.push(11)
s.push(12)
print(s.size())
print(s.isEmpty())
s.push(7)
print(s.pop())
print(s.pop())
print(s.pop())
print(s.size())

File: Stack/Dart/code.dart
class Stack {
  num capacity;
  List storage;

  Stack({capacity = double.infinity}) {
    this.capacity = capacity;
    this.storage = [];
  }

  num get peek => this.storage[this.size - 1];

  num get size => this.storage.length;

  bool get isEmpty => this.storage.isEmpty;

  push(item) {
    if (this.size == this.capacity) {
      throw new RangeError(
          "Stack has reached max capacity, you cannot add more items");
    }
    this.storage.add(item);
  }

  pop() {
    if (!this.storage.isEmpty) {
      return this.storage.removeLast();
    }
  }

  @override
  String toString() {
    return this.storage.toString();
  }
}

main(List<String> args) {
  Stack stack = Stack(capacity: 10);
  print(stack.isEmpty);
  stack.push(10);
  stack.push(11);
  stack.push(12);
  print(stack.isEmpty);
  stack.push(11);
  stack.push(4);
  stack.push(11);
  stack.push(1);
  stack.push(65);
  stack.push(5);
  stack.push(10);
  print(stack.size);
  // stack.push(10);
  print(stack);
}
